modules/apps/site/site-admin-web/src/main/resources/META-INF/resources/site_settings/pages.jsp

```diff
@@ -68,177 +68,181 @@ if (!LanguageUtil.isInheritLocales(siteGoup.getGroupId()) && !siteAdminConfigura
+			<c:choose>
+				<c:when test="<%= (privateLayoutSetPrototype != null) && !siteGoup.isStaged() && hasUnlinkLayoutSetPrototypePermission %>">
+					<c:if test="<%= disableLayoutSetPrototypeInput %>">
+						<div class="alert alert-info">
+							<liferay-ui:message key="you-cannot-enable-the-propagation-of-changes-because-you-modified-the-display-settings-of-this-site" />
+						</div>
+					</c:if>
 
-				<div class="<%= privateLayoutSetPrototypeLinkEnabled ? "" : "hide" %>" id="<portlet:namespace />privateLayoutSetPrototypeMergeAlert">
+					<aui:input disabled="<%= disableLayoutSetPrototypeInput %>" inlineLabel="right" label='<%= LanguageUtil.format(request, "enable-propagation-of-changes-from-the-site-template-x", HtmlUtil.escape(privateLayoutSetPrototype.getName(locale)), false) %>' labelCssClass="simple-toggle-switch" name="privateLayoutSetPrototypeLinkEnabled" type="toggle-switch" value="<%= privateLayoutSetPrototypeLinkEnabled %>" />
 
-					<%
-					request.setAttribute("edit_layout_set_prototype.jsp-groupId", String.valueOf(siteGoup.getGroupId()));
-					request.setAttribute("edit_layout_set_prototype.jsp-layoutSet", privateLayoutSet);
-					request.setAttribute("edit_layout_set_prototype.jsp-layoutSetPrototype", privateLayoutSetPrototype);
-					request.setAttribute("edit_layout_set_prototype.jsp-redirect", currentURL);
-					%>
+					<div class="<%= privateLayoutSetPrototypeLinkEnabled ? "" : "hide" %>" id="<portlet:namespace />privateLayoutSetPrototypeMergeAlert">
 
-					<liferay-util:include page="/layout_set_merge_alert.jsp" servletContext="<%= application %>" />
-				</div>
-			</c:when>
-			<c:when test="<%= privateLayoutSetPrototype != null %>">
-				<liferay-ui:message arguments="<%= HtmlUtil.escape(privateLayoutSetPrototype.getName(locale)) %>" key="these-pages-are-linked-to-site-template-x" translateArguments="<%= false %>" />
-
-				<aui:input name="privateLayoutSetPrototypeLinkEnabled" type="hidden" value="<%= privateLayoutSetPrototypeLinkEnabled %>" />
-			</c:when>
-		</c:choose>
-	</c:otherwise>
-</c:choose>
+						<%
+						request.setAttribute("edit_layout_set_prototype.jsp-groupId", String.valueOf(siteGoup.getGroupId()));
+						request.setAttribute("edit_layout_set_prototype.jsp-layoutSet", privateLayoutSet);
+						request.setAttribute("edit_layout_set_prototype.jsp-layoutSetPrototype", privateLayoutSetPrototype);
+						request.setAttribute("edit_layout_set_prototype.jsp-redirect", currentURL);
+						%>
+
+						<liferay-util:include page="/layout_set_merge_alert.jsp" servletContext="<%= application %>" />
+					</div>
+				</c:when>
+				<c:when test="<%= privateLayoutSetPrototype != null %>">
+					<liferay-ui:message arguments="<%= HtmlUtil.escape(privateLayoutSetPrototype.getName(locale)) %>" key="these-pages-are-linked-to-site-template-x" translateArguments="<%= false %>" />
+
+					<aui:input name="privateLayoutSetPrototypeLinkEnabled" type="hidden" value="<%= privateLayoutSetPrototypeLinkEnabled %>" />
```

Comment:
This line adds a form input. Please check:

* The input is validated on the processing action

* The backend is protected agains value forging



Refer to documentation: